### PR TITLE
Plant line list in submission

### PR DESCRIPTION
**Pull request by kammerer**
This PR introduces the following:

1. Allows to pick existing PLs in 3rd step
2. Allows to define new PLs in 3rd step
3. Stores submitted data as 'regular' objects (PP, PLs, PPLs)

Surely there is a lot of places which need work but I think it is better to address these in separate issues.

### CODE REVIEW

File: app/assets/javascripts/jquery.scrollTo.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
+/*!
```
Comment (Nuanda) [inline app/assets/javascripts/jquery.scrollTo.js]: Shouldn't we put external libraries to vendor/assets/javascripts ?

File: app/assets/javascripts/submissions.coffee [hunk 2]
```diff
@@ -16,50 +16,76 @@ plantLineSelectOptions = ->
...
+    $input = $("<input type='hidden' name='submission[content][new_plant_lines][][" + attr + "]' />")
+    $input.val(val)
+    $input.appendTo($container)
+  )
 
-switchTarget = (button) ->
-  markActiveButton(button)
-  activateTargetFields(button)
+newPlantLineForListContainerId = (plant_line_name) ->
+  'new-plant-line-' + plant_line_name.split(/\s+/).join('-').toLowerCase()
```
Comment (Nuanda) [inline app/assets/javascripts/submissions.coffee]: Are that complex processing required due to not having a proper integer id for a PL? This is why we need to rely on plant_line_name for identification?

File: app/models/plant_line.rb [hunk 3]
```diff
@@ -45,6 +45,9 @@ def self.grid_data(filter)
       .pluck(*columns)
   end
 
+  def self.genetic_statuses
+    order('genetic_status').pluck('DISTINCT genetic_status').reject(&:blank?).reject { |s| ['none', 'not applicable', 'unspecified'].include?(s) }
```
Comment (Nuanda) [inline app/models/plant_line.rb]: Well, I'm not a 80-char zealot ;) but this is just a bit too much. Try to wrap it.

File: app/views/submissions/steps/_step03.html.haml [hunk 4]
```diff
@@ -17,4 +17,64 @@
...
+        %p.help-block Optional
+
+      .action-buttons
+        %button.cancel-new-plant-line-for-list{type: 'button'} Cancel
+        %button.add-new-plant-line-for-list{type: 'button'} Confirm
+
   = render partial: 'submissions/steps/buttons', locals: { f: f, submission: submission }
+
+  - @content.new_plant_lines.each do |plant_line|
+    %div{id: "new-plant-line-#{plant_line.plant_line_name.split.join('-').downcase}"}
```
Comment (Nuanda) [inline app/views/submissions/steps/_step03.html.haml]: Again the "plant_line_name as a unique id" problem?

File: app/views/submissions/steps/_step03.html.haml [hunk 5]
```diff
@@ -17,4 +17,64 @@
...
+      = hidden_field_tag "submission[content][new_plant_lines][][plant_line_name]", plant_line.plant_line_name
+      = hidden_field_tag "submission[content][new_plant_lines][][taxonomy_term]", plant_line.taxonomy_term
+      = hidden_field_tag "submission[content][new_plant_lines][][common_name]", plant_line.common_name
+      = hidden_field_tag "submission[content][new_plant_lines][][previous_line_name]", plant_line.previous_line_name
+      = hidden_field_tag "submission[content][new_plant_lines][][comments]", plant_line.comments
+      = hidden_field_tag "submission[content][new_plant_lines][][genetic_status]", plant_line.genetic_status
```
Comment (Nuanda) [inline app/views/submissions/steps/_step03.html.haml]: Have you considered putting that in a array, with reflection 'send' call on plant_line for each element?
Like:
%w(plant_line_name taxonomy_term ...).each do |attr|
 = hidden_field_tag "submission[content][new_plant_lines][][#{attr}]", plant_line.send(attr)

File: app/services/submission/plant_population_finalizer.rb [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    %i[female_parent_line male_parent_line].each do |parent_line_attr|
+      if parent_line = PlantLine.find_by(plant_line_name: submission.content.step03[parent_line_attr])
+        attrs.merge!(parent_line_attr => parent_line)
+      end
+    end
+
+    attrs.merge!(submission.content.step01.to_h)
+    attrs.merge!(submission.content.step02.to_h)
+    attrs.merge!(submission.content.step04.to_h)
+    attrs.delete(:name) # FIXME change :name to :plant_population_name in the form
```
Comment (Nuanda) [inline app/services/submission/plant_population_finalizer.rb]: I don't understand that thing. What really should be changed in the model and how?

**Comment by Nuanda** [comment]
Made some minor remarks.

File: app/assets/javascripts/jquery.scrollTo.js (hunk 1 above)
Comment (kammerer) [inline app/assets/javascripts/jquery.scrollTo.js]: Makes sense

File: app/assets/javascripts/submissions.coffee (hunk 2 above)
Comment (kammerer) [inline app/assets/javascripts/submissions.coffee]: No, this code handles PLs added during the submission, they are not stored as regular PL records.

File: app/models/plant_line.rb (hunk 3 above)
Comment (kammerer) [inline app/models/plant_line.rb]: Ok. BTW 'none', 'not applicable', etc are already a thing of the past?

File: app/services/submission/plant_population_finalizer.rb (hunk 6 above)
Comment (kammerer) [inline app/services/submission/plant_population_finalizer.rb]: Not in the model, in the form class. I used `name` property but using `plant_population_name` would make things slightly simpler.
On the other hand changing `PlantPopulation#plant_population_name` to `#name` in the model would make sense to.

File: app/models/plant_line.rb (hunk 3 above)
Comment (Nuanda) [inline app/models/plant_line.rb]: They should be.

tg

**Pull request merged**